src/tt.cpp

```diff
@@ -65,9 +117,47 @@ uint8_t TTEntry::relative_age(const uint8_t generation8) const {
     // is needed to keep the unrelated lowest n bits from affecting
     // the result) to calculate the entry age correctly even after
     // generation8 overflows into the next cycle.
+    return (GENERATION_CYCLE + generation8 - genBound8)
+         & GENERATION_MASK;
+}
+
+
+// TTWriter is a very thin wrapper around the pointer but hey it does its job
+TTWriter::TTWriter(TTEntry* tte) : entry(tte) {}
+
+void TTWriter::write(Key k, Value v, bool pv, Bound b, Depth d, Move m, Value ev, uint8_t generation8) {
+    // boiler plate delegation is no fun
+    entry->save(k, v, pv, b, d, m, ev, generation8);
+}
+
+
+// A TranspositionTable is an array of Cluster, of size clusterCount. Each cluster consists of ClusterSize number
+// of TTEntry. Each non-empty TTEntry contains information on exactly one position. The size of a Cluster should
+// divide the size of a cache line for best performance, as the cacheline is prefetched when possible.
+
+static constexpr int ClusterSize = 3;
+
+struct Cluster {
+    TTEntry entry[ClusterSize];
+    char    padding[2];  // Pad to 32 bytes
+};
+
+// This can be removed when hacking `Cluster`, but is (so far) necessary for best performance.
```

Comment:
unneeded comment

Reply:
I've personally seen Robby confused by this, and he's representative of a very broad swath of readers -- this comment would have helped him reach his goal. Are you sure you want to remove this?

Reply:
Since I have been mentioned here: my impression was the assertion triggered because my fiddling with the TT entry would break sf (I had tried to also store the FEN in the TT, iirc).

Reply:
It doesn't break SF, just lessens efficiency.

I've pushed a version with no comment but a mildly tweaked assert message.